### PR TITLE
ref(replay): add docs about custom breadcrumbs

### PR DESCRIPTION
**Pull request by michellewzhang**
closes https://github.com/getsentry/sentry-docs/issues/12178

### CODE REVIEW

**Review by aliu39 — APPROVED** [review]

**Review by billyvg — COMMENTED** [review]
LGTM -- should we talk about the title as well? 

https://github.com/getsentry/sentry/blob/master/static/app/utils/replays/getFrameDetails.tsx#L458-L471

**Comment by michellewzhang** [comment]
> LGTM -- should we talk about the title as well?
> 
> https://github.com/getsentry/sentry/blob/master/static/app/utils/replays/getFrameDetails.tsx#L458-L471

the title will always be "custom" since the category is set to be `custom` in `hydrateBreadcrumbs.tsx`:

https://github.com/getsentry/sentry/blob/20d585fc120c3c0d086c353ad04d04c8bf3d4917/static/app/utils/replays/hydrateBreadcrumbs.tsx#L28-L39

**Comment by billyvg** [comment]
> getsentry/sentry@`master`/static/app/utils/replays/getFrameDetails.tsx#L458-L471

@michellewzhang Isn't it `category` -> `defaultTitle()` -> `'custom'`? (And then `defaultTitle()` again in `<BreadcrumbItem>`?

File: docs/platforms/javascript/common/session-replay/configuration.mdx [hunk 1]
```diff
@@ -158,3 +158,19 @@ replayIntegration({
...
+
+In Replay Details, the breadcrumb `message` will take precedence over the `data` property. If the `message` property is set, then the string will be displayed like this:
+
+![Custom breadcrumb example with message](./img/message-custom-breadcrumb.png)
+
+If `message` is not set, then the breadcrumb will display the `data` property by default. For example:
+
+![Custom breadcrumb example with data](./img/data-custom-breadcrumb.png)
+
+The title of the breadcrumb will be the breadcrumb `category` that you set. If no `category` is set, then we will use the breadcrumb `message`. If that doesn't exist, we'll use the `description`. The last fallback with be a default title of "Custom".
```
Comment (billyvg) [inline docs/platforms/javascript/common/session-replay/configuration.mdx:176]: 😭

**Review by billyvg — APPROVED** [review]

File: docs/platforms/javascript/common/session-replay/configuration.mdx (hunk 1 above)
Comment (michellewzhang) [reply docs/platforms/javascript/common/session-replay/configuration.mdx:176]: 💀

**Pull request merged**